source/apiVolontaria/volunteer/models.py

```diff
@@ -76,3 +86,84 @@ def save(self, *args, **kwargs):
 
     def __str__(self):
         return '{}, {}'.format(self.name, str(self.address))
+
+
+class Event(models.Model):
+    """
+
+    This class represents the Event model.
+
+    """
+    class Meta:
+        verbose_name_plural = 'Events'
+
+    start_date = models.DateTimeField(
+        verbose_name="Begin date",
+        blank=False,
+        null=False,
+    )
+
+    end_date = models.DateTimeField(
+        verbose_name="End date",
+        blank=False,
+        null=False,
+    )
+
+    nb_volunteers_needed = models.PositiveIntegerField(
+        verbose_name="Number of volunteer needed",
+        default=0,
+    )
+    nb_volunteers_standby_needed = models.PositiveIntegerField(
+        verbose_name="Number of volunteers standby needed",
+        default=0,
+    )
+
+    volunteers = models.ManyToManyField(
+        User,
+        verbose_name="Volunteers",
+        related_name="events",
+        blank=True,
+    )
+
+    volunteers_standby = models.ManyToManyField(
+        User,
+        verbose_name="Volunteers standby",
+        related_name="events_standby",
+        blank=True,
+    )
+
+    cell = models.ForeignKey(
+        Cell,
+        verbose_name="Cell",
+        blank=False,
+    )
+
+    cycle = models.ForeignKey(
+        Cycle,
+        verbose_name="Cycle",
+        blank=False,
+    )
+
+    task_type = models.ForeignKey(
+        TaskType,
+        verbose_name="Task type",
+        blank=False,
+    )
+
+    def save(self, *args, **kwargs):
+        super(Event, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return '{0}, {1} - {2}'.format(
+            self.cell,
+            str(self.start_date),
+            str(self.end_date),
+        )
+
+    @property
+    def is_expired(self):
+        return self.end_date <= timezone.now()
+
+    @property
+    def is_active(self):
+        return self.cycle.is_active
```

Comment:
Is it a good idea that the Event has the same `activity state` as its cycle? What if we want a `Cycle` to be active, but not the `Event` inside that cycle?

What is the disctinction between `is_expired` and `is_active`? Do we really need both?

Reply:
`is_active()` allow us to know if this event need to be displayed for all people. Actually requirements say we don't need to display event of a cycle if this cycle is not active, so this function just return the result of the cycle function. If an event is not active, just admin can access it to manage it.

I've created  `event.is_active()` to allow us to modify this functionality without a big refactor. Today it depend of the cycle, tomorrow we can change it here.

---

`is_expired()` allow us to know if the event date is already pasted, it's just to help us filter each event.